draft-ietf-tls-tls13.md

```diff
@@ -3804,15 +3843,30 @@ then delivered to higher-level clients.
 
 TLS records are typed, which allows multiple higher-level protocols to
 be multiplexed over the same record layer. This document specifies
-three content types: handshake, application data, and alert.
+four content types: handshake, application data, alert, and
+change_cipher_spec, with the latter being used only for
+compatibility purposes (see {{middlebox}}).
+
+An implementation may receive an unencrypted record of type
+change_cipher_spec consisting of the single byte value 0x01 at any
+time during the handshake and MUST simply drop it without further
+processing.  Note that this record may appear at a point at the
+handshake where the implementation is expecting protected records
+and so it is necessary to detect this
+condition prior to attempting to deprotect the record. An
+implementation which receives any other change_cipher_spec value or
+which receives a protected change_cipher_spec record MUST abort the
+handshake with an "illegal_parameter" alert. After the handshake is
+complete, change_cipher_spec MUST be treated as an unexpected record
+type.
```

Comment:
I wonder how much more complicated implementations will have to be to reject CCS post-handshake (as opposed to ignoring it always).

Reply:
Not very it seems.